hie-base/Haskell/Ide/Engine/PluginTypes.hs

```diff
@@ -1,3 +1,7 @@
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
Can we move the singleton definitions into their own module, which has
UndecidableInstances, and the rest of the code does not?

On Thu, Jan 7, 2016 at 6:08 PM, Moritz Kiefer <notifications@github.com>
wrote:

> In hie-base/Haskell/Ide/Engine/PluginTypes.hs
> <https://github.com/haskell/haskell-ide-engine/pull/152#discussion_r49089527>
> :
>
> > @@ -1,3 +1,7 @@
> > +{-# LANGUAGE InstanceSigs #-}
> > +{-# LANGUAGE UndecidableInstances #-}
>
> Which would mean that we also need to implement our own concatMap type
> family or inline that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/haskell/haskell-ide-engine/pull/152/files#r49089527>.
>